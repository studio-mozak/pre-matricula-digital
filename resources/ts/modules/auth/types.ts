export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
