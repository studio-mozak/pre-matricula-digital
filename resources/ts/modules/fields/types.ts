import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
