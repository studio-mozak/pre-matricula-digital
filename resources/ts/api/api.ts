import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
