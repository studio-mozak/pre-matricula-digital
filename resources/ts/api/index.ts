export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
