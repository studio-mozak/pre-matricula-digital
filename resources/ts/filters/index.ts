import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
