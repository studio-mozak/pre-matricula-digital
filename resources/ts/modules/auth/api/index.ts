/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
