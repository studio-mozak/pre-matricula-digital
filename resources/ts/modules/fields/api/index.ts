/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
