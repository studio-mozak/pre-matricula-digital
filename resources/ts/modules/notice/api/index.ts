/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
