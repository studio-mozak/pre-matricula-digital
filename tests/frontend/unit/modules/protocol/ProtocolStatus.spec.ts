import {
  ProtocolStatusPreRegistration,
  ProtocolStatusReturnToWait,
} from '@/modules/protocol/types';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { Protocol as ProtocolApi } from '@/modules/protocol/api';
import ProtocolStatus from '@/modules/protocol/components/ProtocolStatus.vue';
import { createTestingPinia } from '@pinia/testing';
import { useGeneralStore } from '@/store/general';
import VTooltip from 'v-tooltip';
import filters from '@/filters';
import { expect, vi } from 'vitest';
import { Nullable } from '@/types';

vi.mock('@/modules/protocol/api');
vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRoute: () => ({
    params: {
      id: '000AAA',
    },
  }),
}));
vi.mock('vee-validate');

const preregistrationMock: ProtocolStatusPreRegistration = {
  classroom: {
    grade: {
      name: 'string',
    },
    name: 'string',
    period: {
      name: 'string',
    },
  },
  id: 'string',
  observation: 'string',
  parent: {
    id: 1,
    protocol: 'string',
    code: 'string',
    type: 'string',
    date: 'string',
    position: 1,
    school: {
      id: 1,
      name: 'string',
      area_code: 'string',
      phone: 'string',
    },
    process: {
      showPriorityProtocol: false,
    },
  },
  position: 1,
  process: {
    showPriorityProtocol: false,
    blockIncompatibleAgeGroup: false,
    forceSuggestedGrade: false,
    grades: [],
  },
  school: {
    area_code: 47,
    name: 'string',
    phone: 'string',
  },
  stage: {
    observation: 'string',
  },
  status: 'ACCEPTED',
  student: {
    dateOfBirth: 'string',
    initials: 'string',
  },
  type: 'string',
  waiting: {
    id: 1,
    protocol: 'string',
    code: 'string',
    type: 'string',
    date: 'string',
    position: 1,
    school: {
      id: 1,
      name: 'string',
      area_code: 'string',
      phone: 'string',
    },
    process: {
      showPriorityProtocol: false,
    },
  },
};

const returnToWaitResponseMock: ProtocolStatusReturnToWait = {
  errors: false,
  keepOnTheWaitingList: {
    id: '1',
  },
};

const onMounted = async (
  preregistration: Nullable<ProtocolStatusPreRegistration>
) => {
  vi.mocked(ProtocolApi.show).mockResolvedValueOnce(preregistration);

  const wrapper = mount(ProtocolStatus, {
    plugins: [createTestingPinia()],
    global: {
      stubs: [
        'protocol-consult',
        'router-link',
        'protocol-confirm-keep-in-waiting-list',
      ],
      directives: {
        tooltip: VTooltip,
      },
      provide: {
        $filters: filters,
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  return wrapper;
};

describe('ProtocolStatus', () => {
  let wrapper: VueWrapper<InstanceType<typeof ProtocolStatus>>;

  beforeEach(async () => {
    wrapper = await onMounted(preregistrationMock);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('check if I have 1 protocol-consult, pre-registration-position and 3 modals', () => {
    const protocolConsult = wrapper.findComponent({
      name: 'protocol-consult',
    });

    expect(protocolConsult).toBeTruthy();

    const preRegistrationPosition = wrapper.findComponent({
      name: 'pre-registration-position',
    });

    expect(preRegistrationPosition).toBeTruthy();

    const modals = wrapper.findAllComponents({
      name: 'modal',
    });

    expect(modals).toBeTruthy();
    expect(modals.length).toBe(1);
  });

  test('the component has the correct data', () => {
    expect(wrapper.vm.preregistration).toMatchObject(preregistrationMock);
    expect(wrapper.vm.notFound).toBeFalsy();
  });

  test('when I click in the "Mais informações" button, the modal is open', async () => {
    const button = wrapper.find('[data-test="btn-more-info"]');

    await button.trigger('click');

    expect(wrapper.vm.showModal).toBeTruthy();
  });

  test('"Continuar na lista" button, should not be shown if isAthenticated is false', async () => {
    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'IN_CONFIRMATION';

    wrapper = await onMounted(preregistrationMockNewStatus);

    const store = useGeneralStore();

    store.auth.authenticated = false;

    await wrapper.vm.$nextTick();

    const button = wrapper.find('[data-test="btn-continue-on-the-list"]');

    expect(button.exists()).toBeFalsy();
  });

  test('when I click in the "Continuar na lista" button, with status IN_CONFIRMATION the modal is open and must to trigger returnToWait function', async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'IN_CONFIRMATION';

    wrapper = await onMounted(preregistrationMockNewStatus);

    const store = useGeneralStore();

    store.auth.authenticated = true;

    await wrapper.vm.$nextTick();

    const button = wrapper.find('[data-test="btn-continue-on-the-list"]');

    await button.trigger('click');

    expect(wrapper.vm.showModalConfirmKeepInWaitingList).toBeTruthy();
  });

  test('when I click in the "Continuar na lista" button, with status IN_CONFIRMATION the modal is open and must to trigger returnToWait function and return error', async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockRejectedValueOnce({
      response: {
        data: {
          ...returnToWaitResponseMock,
          errors: true,
        },
      },
    });

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'IN_CONFIRMATION';

    wrapper = await onMounted(preregistrationMockNewStatus);

    await wrapper.vm.$nextTick();

    const store = useGeneralStore();

    store.auth.authenticated = true;

    await wrapper.vm.$nextTick();

    const button = wrapper.find('[data-test="btn-continue-on-the-list"]');

    await button.trigger('click');

    expect(wrapper.vm.showModalConfirmKeepInWaitingList).toBeTruthy();
  });

  test("If the status of preregistration is WAITING, the alert label must be 'Em espera'", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'WAITING';

    wrapper = await onMounted(preregistrationMockNewStatus);

    const label = wrapper.find(
      '[class="alert badge-yellow text-center text-uppercase"]'
    );

    expect(label.text()).toBe('Em espera');
  });

  test("If the status of preregistration is REJECTED, the alert label must be 'Indeferido'", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'REJECTED';

    wrapper = await onMounted(preregistrationMockNewStatus);

    const label = wrapper.find(
      '[class="alert badge-red text-center text-uppercase"]'
    );

    expect(label.text()).toBe('Indeferido');
  });

  test("If the status of preregistration is SUMMONED, the alert label must be 'Responsáveis Convocados'", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'SUMMONED';

    wrapper = await onMounted(preregistrationMockNewStatus);

    const label = wrapper.find(
      '[class="alert badge-purple text-center text-uppercase"]'
    );

    expect(label.text()).toBe('Responsáveis Convocados');
  });

  test("If showPriorityProtocol is true and status is WAITING, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'WAITING';

    preregistrationMockNewStatus.process.showPriorityProtocol = true;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeTruthy();
  });

  test("If showPriorityProtocol is true and status is ACCEPTED, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'ACCEPTED';

    preregistrationMockNewStatus.process.showPriorityProtocol = true;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeTruthy();
  });

  test("If showPriorityProtocol is true and status is REJECTED, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'REJECTED';

    preregistrationMockNewStatus.process.showPriorityProtocol = true;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeTruthy();
  });

  test("If showPriorityProtocol is true and status is SUMMONED, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'SUMMONED';

    preregistrationMockNewStatus.process.showPriorityProtocol = true;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeTruthy();
  });

  test("If showPriorityProtocol is false and status is WAITING, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'WAITING';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeFalsy();
  });

  test("If showPriorityProtocol is false and status is ACCEPTED, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'ACCEPTED';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeFalsy();
  });

  test("If showPriorityProtocol is false and status is REJECTED, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'REJECTED';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeFalsy();
  });

  test("If showPriorityProtocol is false and status is SUMMONED, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'SUMMONED';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position"]'
    );

    expect(preRegistrationPosition.exists()).toBeFalsy();
  });

  test("If process is waiting another process, and this waiting process is showing priority protocol, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'WAITING';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    if (preregistrationMockNewStatus.waiting.process) {
      preregistrationMockNewStatus.waiting.process.showPriorityProtocol = true;
    }

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position-waiting"]'
    );

    expect(preRegistrationPosition.exists()).toBeTruthy();
  });

  test("If process is waiting another process, and this waiting process isn't showing priority protocol, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'WAITING';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    if (preregistrationMockNewStatus.waiting.process) {
      preregistrationMockNewStatus.waiting.process.showPriorityProtocol = false;
    }

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position-waiting"]'
    );

    expect(preRegistrationPosition.exists()).toBeFalsy();
  });

  test("If process have parent, and this parent process is showing priority protocol, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'WAITING';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    if (preregistrationMockNewStatus.parent.process) {
      preregistrationMockNewStatus.parent.process.showPriorityProtocol = true;
    }

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position-parent"]'
    );

    expect(preRegistrationPosition.exists()).toBeTruthy();
  });

  test("If process have parent, and this parent process isn't showing priority protocol, the 'pre-registration-position' component must be mounted", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'WAITING';

    preregistrationMockNewStatus.process.showPriorityProtocol = false;

    if (preregistrationMockNewStatus.parent.process) {
      preregistrationMockNewStatus.parent.process.showPriorityProtocol = false;
    }

    wrapper = await onMounted(preregistrationMockNewStatus);

    const preRegistrationPosition = wrapper.find(
      '[data-test="preregistration-position-parent"]'
    );

    expect(preRegistrationPosition.exists()).toBeFalsy();
  });

  test("If process status is 'IN_CONFIRMATION' and I click in 'Mais informações' button, the showModal variable must be true", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      returnToWaitResponseMock
    );

    const preregistrationMockNewStatus = preregistrationMock;

    preregistrationMockNewStatus.status = 'IN_CONFIRMATION';

    wrapper = await onMounted(preregistrationMockNewStatus);

    expect(wrapper.vm.showModal).toBeFalsy();

    const btnMoreInfo = wrapper.find('[data-test="btn-more-info"]');

    await btnMoreInfo.trigger('click');

    expect(wrapper.vm.showModal).toBeTruthy();
  });

  test("If 'notFound' variable is true and preregistration is null, properly text must be rendered", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      null as unknown as ProtocolStatusReturnToWait
    );

    wrapper = await onMounted(null);

    expect(wrapper.vm.preregistration).toBeNull();

    expect(wrapper.vm.notFound).toBeTruthy();

    const notFoundText = wrapper.find('[data-test="not-found-text"]');

    expect(notFoundText.exists()).toBeTruthy();

    expect(notFoundText.text()).toBe('Nenhum protocolo foi encontrado.');
  });

  test("If 'notFound' variable is false and preregistration is null, loader component must be rendered", async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce(
      null as unknown as ProtocolStatusReturnToWait
    );

    wrapper = await onMounted(null);

    wrapper.vm.notFound = false;

    await wrapper.vm.$nextTick();

    expect(wrapper.vm.preregistration).toBeNull();

    expect(wrapper.vm.notFound).toBeFalsy();

    const loader = wrapper.find('[data-test="loader-content"]');

    expect(loader.exists()).toBeTruthy();
  });
});
