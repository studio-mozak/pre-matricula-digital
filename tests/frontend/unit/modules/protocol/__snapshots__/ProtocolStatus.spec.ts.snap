// Vitest Snapshot v1

exports[`ProtocolStatus > should match snapshot 1`] = `
<div
  class="container mt-4 pt-lg-4"
  data-v-170b8f14=""
>
  <div
    class="row mt-3"
    data-v-170b8f14=""
  >
    <div
      class="col-12 col-md-10 offset-md-1"
      data-v-170b8f14=""
    >
      <h2
        class="font-muli-20-primary mb-4 text-center"
        data-v-170b8f14=""
      >
         Resultado de consulta 
      </h2>
      <p
        class="text-center"
        data-v-170b8f14=""
      >
         Certifique-se que as iniciais do nome e a data de nascimento são compatíveis com as do(a) aluno(a) sendo pesquisado. Se os dados estiverem incorretos verifique se o número do protocolo está correto e tente novamente. 
      </p>
    </div>
  </div>
  <div
    class="row mt-3"
    data-v-170b8f14=""
  >
    <div
      class="col-12 col-md-10 offset-md-1"
      data-v-170b8f14=""
    >
      <protocol-consult-stub
        data-v-170b8f14=""
        title="Consulta de inscrição"
      />
    </div>
  </div>
  <div
    class="row mt-3"
    data-v-170b8f14=""
  >
    <div
      class="col-12 col-md-10 offset-md-1"
      data-v-170b8f14=""
    >
      <div
        class="m-auto pt-4"
        data-v-170b8f14=""
      >
        <div
          class="protocol card p-3 m-auto"
          data-v-170b8f14=""
        >
          <div
            data-v-170b8f14=""
          >
            <div
              data-v-170b8f14=""
            >
              <div
                class="alert badge-green text-center text-uppercase"
                data-v-170b8f14=""
              >
                 Deferido 
              </div>
            </div>
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
            <dl
              class="mb-0"
              data-v-170b8f14=""
            >
              <dt
                data-v-170b8f14=""
              >
                Aluno(a)
              </dt>
              <dd
                data-v-170b8f14=""
              >
                string (Data inválida) 
              </dd>
              <!--v-if-->
              <dt
                data-v-170b8f14=""
              >
                Escola
              </dt>
              <dd
                data-v-170b8f14=""
              >
                <p
                  class="mb-0"
                  data-v-170b8f14=""
                >
                  string
                </p>
                <p
                  class="mb-0"
                  data-v-170b8f14=""
                >
                   Telefone: (47) string
                </p>
              </dd>
              <dt
                data-v-170b8f14=""
              >
                Tipo
              </dt>
              <dd
                data-v-170b8f14=""
              >
                <span
                  class="badge badge-blue"
                  data-v-170b8f14=""
                >
                   Pré-matrícula 
                </span>
              </dd>
            </dl>
            <div
              class="mt-4"
              data-v-170b8f14=""
            >
              <div
                class="alert badge-gray"
                data-v-170b8f14=""
              >
                <p
                  data-v-170b8f14=""
                >
                  Este aluno também está em lista de espera:
                </p>
                <dl
                  class="mb-0"
                  data-v-170b8f14=""
                >
                  <dt
                    data-v-170b8f14=""
                  >
                    Protocolo
                  </dt>
                  <dd
                    data-v-170b8f14=""
                  >
                    <router-link-stub
                      data-v-170b8f14=""
                      to="[object Object]"
                    />
                  </dd>
                  <!--v-if-->
                  <dt
                    data-v-170b8f14=""
                  >
                    Escola
                  </dt>
                  <dd
                    data-v-170b8f14=""
                  >
                    <p
                      class="mb-0"
                      data-v-170b8f14=""
                    >
                      string
                    </p>
                    <p
                      class="mb-0"
                      data-v-170b8f14=""
                    >
                       Telefone: (string)
string
                    </p>
                  </dd>
                </dl>
              </div>
            </div>
            <div
              class="mt-4"
              data-v-170b8f14=""
            >
              <div
                class="alert badge-gray"
                data-v-170b8f14=""
              >
                <p
                  data-v-170b8f14=""
                >
                  Este(a) aluno(a) também possui uma pré-matrícula:
                </p>
                <dl
                  class="mb-0"
                  data-v-170b8f14=""
                >
                  <dt
                    data-v-170b8f14=""
                  >
                    Protocolo
                  </dt>
                  <dd
                    data-v-170b8f14=""
                  >
                    <router-link-stub
                      data-v-170b8f14=""
                      to="[object Object]"
                    />
                  </dd>
                  <!--v-if-->
                  <dt
                    data-v-170b8f14=""
                  >
                    Escola
                  </dt>
                  <dd
                    data-v-170b8f14=""
                  >
                    <p
                      class="mb-0"
                      data-v-170b8f14=""
                    >
                      string
                    </p>
                    <p
                      class="mb-0"
                      data-v-170b8f14=""
                    >
                       Telefone: (string)
string
                    </p>
                  </dd>
                </dl>
              </div>
            </div>
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 mt-4"
              data-test="btn-more-info"
              data-v-170b8f14=""
              role="button"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Mais informações
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
            <!--v-if-->
          </div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>
`;
